lib/src/lib/table/components/table/table.component.ts

```diff
@@ -349,6 +362,22 @@ export class Lab900TableComponent<T extends object = object> implements OnChange
     this.tableCellsFiltered.emit(tableCells);
   }
 
+  public onActiveTabChange(id: TabId): void {
+    this.selection.clear();
+    const previousTableTab = this.tableTabs.find((t) => t.id === this.activeTabId);
+    const activeTableTab = this.tableTabs.find((t) => t.id === id);
+
+    if (activeTableTab?.tableCells?.length) {
+      this.tableCells = activeTableTab.tableCells;
+      // load the original tableCells if the new active tab hasn't got any specific table cells but the previous tab had
+    } else if (previousTableTab?.tableCells?.length && !activeTableTab?.tableCells?.length && this.originalCells?.length) {
+      this.tableCells = this.originalCells;
+    }
+
+    this.activeTabId = id;
+    this.activeTabIdChange.emit(id);
```

Comment:
Should the new Id be emitted, even if you went in the `else` statement above?

Reply:
You'll always want to emit an event to be able to do other actions, like loading specific data for that tab from the parent components.